hadoop-plugin/src/main/groovy/com/linkedin/gradle/azkaban/yaml/YamlWorkflow.groovy

```diff
@@ -0,0 +1,180 @@
+   * @param parentScope The parent scope of the workflow
+   * @param subflow Boolean regarding whether or not the workflow is a subflow
+   */
+  YamlWorkflow(Workflow workflow, NamedScope parentScope, boolean isSubflow) {
+    // Include name/type in yaml only if the workflow is a subflow
+    name = isSubflow ? workflow.name : null;
+    type = isSubflow ? "flow" : null;
+    dependsOn = workflow.parentDependencies.toList();
+    config = buildConfig(workflow, parentScope, isSubflow);
+    nodes = buildNodes(workflow, parentScope);
+  }
+
+  /**
+   * Create the workflow config from the properties in the namespace
+   *
+   * @param workflow The workflow being constructed
+   * @param parentScope The parent scope of the workflow being constructed
+   * @param subflow Boolean regarding whether or not the workflow is a subflow
+   * @return result The map of all properties associated with the workflow
+   */
+  private static Map buildConfig(Workflow workflow, NamedScope parentScope, boolean isSubflow) {
+    Map<String, String> result = [:];
+    // For the root workflow, add all global properties
+    if (!isSubflow) {
+      NamedScope oldParentScope = (NamedScope) parentScope.properties["nextLevel"];
+      oldParentScope.properties["thisLevel"].each { key, val ->
+        // thisLevel contains properties as well as workflows,
+        // so make sure to only parse and include Properties objects
+        if (val.getClass() == Properties) {
+          result << ((Properties) val).buildProperties(oldParentScope)
+        }
+      }
+    }
+    // Build all workflow properties after root properties so if same property is defined
+    // then the workflow property is selected
+    workflow.properties.each { Properties prop ->
+      result << prop.buildProperties(parentScope);
+    }
+    return result;
+  }
+
+  /**
+   * Create the workflow nodes from the jobs/subflows defined in the workflow
+   * Instead of storing YamlJobs/YamlWorkflows themselves, store as maps in order to simplify
+   * yaml output
+   * Do not include LaunchJobs and StartJobs because they aren't needed in Flow 2.0
+   *
+   * @param workflow The workflow being constructed
+   * @param parentScope The parent scope of the workflow being constructed
+   * @return result List of all nodes converted to String Maps
+   */
+  private static List buildNodes(Workflow workflow, NamedScope parentScope) {
+    List result = [];
+
+    // Add all jobs except LaunchJobs and StartJobs
+    workflow.jobsToBuild.each { Job job ->
+      if (job.class != LaunchJob.class && job.class != StartJob.class) {
+        result.add((new YamlJob(job, parentScope)).yamlize());
+      }
+    }
+    // Add all subflows
+    workflow.flowsToBuild.each { Workflow subflow ->
+      result.add((new YamlWorkflow(subflow, subflow.scope, true)).yamlize());
+    }
+    // Remove all LaunchJobs and StartJobs from dependencies of other jobs
+    workflow.jobsToBuild.each { Job job ->
```

Comment:
I think we can keep the lauchJobs and startJobs for now before the new DAG engine feature is ready. I'm ok with checking this part in, but just want to keep a reminder here that the flow execution might not work.

Reply:
Makes sense to me! Thanks for pointing it out. We can refactor this in the future should we want to deploy before the DAG engine is completed.